editor/src/components/canvas/canvas-strategies/strategies/grid-resize-element-ruler-strategy.ts

```diff
@@ -0,0 +1,300 @@
+import type { GridCellCoordBounds } from './grid-cell-bounds'
+import { getGridChildCellCoordBoundsFromCanvas } from './grid-cell-bounds'
+import { getCommandsForGridItemPlacement } from './grid-helpers'
+import { normalizeGridElementPositionAfterResize } from './grid-resize-element-strategy'
+
+export const gridResizeElementRulerStrategy: CanvasStrategyFactory = (
+  canvasState: InteractionCanvasState,
+  interactionSession: InteractionSession | null,
+) => {
+  const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
+  if (selectedElements.length !== 1) {
+    return null
+  }
+
+  const selectedElement = selectedElements[0]
+  const selectedElementMetadata = MetadataUtils.findElementByElementPath(
+    canvasState.startingMetadata,
+    selectedElement,
+  )
+  if (selectedElementMetadata == null) {
+    return null
+  }
+  const isElementInsideGrid = MetadataUtils.isGridItem(
+    canvasState.startingMetadata,
+    selectedElement,
+  )
+  if (!isElementInsideGrid) {
+    return null
+  }
+
+  const selectedElementBounds = MetadataUtils.getFrameInCanvasCoords(
+    selectedElement,
+    canvasState.startingMetadata,
+  )
+  if (selectedElementBounds == null || isInfinityRectangle(selectedElementBounds)) {
+    return null
+  }
+
+  return {
+    id: 'GRID-CELL-RESIZE-RULER-STRATEGY',
+    name: 'Resize Grid Cell (ruler)',
+    descriptiveLabel: 'Resize Grid Cell (ruler)',
+    icon: {
+      category: 'tools',
+      type: 'pointer',
+    },
+    controlsToRender: [
+      {
+        control: GridResizeControls,
+        props: { target: gridContainerIdentifier(selectedElement) },
+        key: `grid-resize-controls-${EP.toString(selectedElement)}`,
+        show: 'always-visible',
+      },
+      controlsForGridPlaceholders(gridItemIdentifier(selectedElement)),
+    ],
+    fitness: onlyFitWhenDraggingThisControl(interactionSession, 'GRID_RESIZE_RULER_HANDLE', 1),
+    apply: () => {
+      if (
+        interactionSession == null ||
+        interactionSession.interactionData.type !== 'DRAG' ||
+        interactionSession.interactionData.drag == null ||
+        interactionSession.activeControl.type !== 'GRID_RESIZE_RULER_HANDLE'
+      ) {
+        return emptyStrategyApplicationResult
+      }
+
+      if (!isFillOrStretchModeAppliedOnAnySide(canvasState.startingMetadata, selectedElement)) {
+        // TODO this should be removed to support resizing cells that are not stretching
+        return emptyStrategyApplicationResult
+      }
+
+      const allCellBounds =
+        selectedElementMetadata.specialSizeMeasurements.parentGridCellGlobalFrames
+
+      if (allCellBounds == null) {
+        return emptyStrategyApplicationResult
+      }
+
+      const bounds = getGridChildCellCoordBoundsFromCanvas(selectedElementMetadata, allCellBounds)
+      if (bounds == null) {
+        return emptyStrategyApplicationResult
+      }
+
+      const columns = allCellBounds[0]
```

Comment:
Doesn't look like we bounds check this array.

Reply:
https://github.com/concrete-utopia/utopia/pull/6689/commits/c344a8439fecdcc914eb97a8a1ebd9c84acc553c